### PR TITLE
fix(Scripts/Temple of AhnQiraj): Moved Obsidian Nullifier script from…

### LINKED ISSUE #12876: [AQ40] Obsidian Nullifier (15312)
### Current Behaviour

Scripted in SAI.
Only targets one player for Drain Mana.
Nullify does no damage.

### Expected Blizzlike Behaviour

Very similar scripting to Obsidian Eradicator
Drain Mana (25671) is cast every 6s at 11 mana users, Drain Mana visual (26639) spell is cast at the same time.
Does not regen mana over time.
At full mana casts Nullify (26552), which reduces players' HP to 1.
Casts Cleave every 6s to 8.4s

Edit: The packs of Anubisath Warders and Obsidian Nullifiers **will respawn if all 3 are not killed together** i.e. if one Nullifier and one Warder are killed but the raid wipes the entire pack will respawn and reset.

### Source

_No response_

### Steps to reproduce the problem

1. `.go c id 15312` with 2 or more mana users
2. see if it does not regenerate mana passively
3. see if it casts Cleave
4. see if it casts Drain Mana along with the visual spell every 6s
5. switch one mana user for one non-mana user, see if Drain Mana is **not** cast on them

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/9ae2b93514b16adf78b27543d0947c694bbb2f29

### Operating system

Windows 10

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
… DB to C++.

Fixes #12876

## Issues Addressed:

- Closes #12876

## Tests Performed:

- Not tested.

## How to Test the Changes:

`.go c id 15312` with 2 or more mana users
see if it does not regenerate mana passively
see if it casts Cleave
see if it casts Drain Mana along with the visual spell every 6s
switch one mana user for one non-mana user, see if Drain Mana is not cast on them

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Nullify isn't consuming the Nullifier's mana and it's regenerating mana out of combat.

**Comment by UltraNix** [comment]
@Gultask Should be okay now.

**Comment by Gultask** [comment]
Works now 👍

**Comment by IntelligentQuantum** [comment]
```
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/scripts/Kalimdor/TempleOfAhnQiraj/temple_of_ahnqiraj.cpp:214:45: fatal error: member access into incomplete type 'const MapReference'
                    if (Player* player = itr->GetSource())
```

**Comment by UltraNix** [comment]
@IntelligentQuantum Should be okay now.

File: src/server/scripts/Kalimdor/TempleOfAhnQiraj/temple_of_ahnqiraj.cpp [hunk 1]
```diff
@@ -183,8 +189,119 @@ class spell_aggro_drones : public SpellScript
...
+        _scheduler.Update(diff,
+            std::bind(&ScriptedAI::DoMeleeAttackIfReady, this));
+    }
+
+private:
+    TaskScheduler _scheduler;
+    std::list<Player*> _targets;
+};
+
+class spell_drain_mana : public SpellScript
```
Comment (pklloveyou) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/temple_of_ahnqiraj.cpp:261]: https://github.com/azerothcore/azerothcore-wotlk/blob/e623e9ef49bdcc986cc32683ada3911820f858a5/src/server/scripts/Kalimdor/RuinsOfAhnQiraj/ruins_of_ahnqiraj.cpp#L147 Their functions are the same, but they will conflict if they exist at the same time

**Comment by UltraNix** [comment]
@IntelligentQuantum Conflicts resolved.

**Comment by IntelligentQuantum** [comment]
@UltraNix Codestyle

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by UltraNix** [comment]
@IntelligentQuantum Done.

**Pull request merged**